components/experimental/src/dimension/provider/extended_currency.rs

```diff
@@ -61,31 +60,9 @@ pub struct CurrencyExtendedDataV1<'data> {
     databake(path = icu_experimental::dimension::provider::extended_currency)
 )]
 #[repr(u8)]
-pub enum Count {
-    /// The CLDR keyword `zero`.
-    Zero = 0,
-    /// The CLDR keyword `one`.
-    One = 1,
-    /// The CLDR keyword `two`.
-    Two = 2,
-    /// The CLDR keyword `few`.
-    Few = 3,
-    /// The CLDR keyword `many`.
-    Many = 4,
-    /// The CLDR keyword `other`.
-    Other = 5,
-    // TODO(younies): revise this for currency
-    /// The explicit 1 case, see <https://www.unicode.org/reports/tr35/tr35-numbers.html#Explicit_0_1_rules>.
-    Explicit1 = 6,
-    // NOTE(egg): No explicit 0, because the compact decimal pattern selection
-    // algorithm does not allow such a thing to arise.
-    // TODO(younies): implment this case.
-    /// The default case.
-    /// NOTE:
-    ///     Used as the default when there is no match.
-    ///     This is also used to replace the most frequently occurring case in all plural rules.
-    Default = 7,
+pub enum CurrencyDisplayNameCount {
+    PluralRules(Count),
 
     /// The display name for the currency.
-    DisplayName = 8,
+    DisplayName,
```

Comment:
Issue: This is in CLDR but it isn't needed for currency formatting, only currency display names

Reply:
By having `DisplayName` as a field, as in the draft PR, https://github.com/unicode-org/icu4x/pull/5389, we are having two issues:

1. We will reserve extra space for `DisplayName` that is not necessary.
2. `serde` has an issue with the life time of `Option<Cow<'data, str>>` as you can see the draft PR.

Reply:
> We will reserve extra space for DisplayName that is not necessary.

`DisplayName` seems to always exist, so not putting it in the `ZeroMap` actually uses less space. You probably don't need to use an `Option`

Reply:
`DisplayName` is not appearing in some locales; therefore, it needs to be optional.

Reply:
There are 11 locale/currency combinations that have plural rules but no display name:
```
bs ARL ("argentinskih pezosa leja")
bs ARM ("argentinskih pezosa monedo nacional")
ca CNX ("dòlars del Banc Popular Xinès")
ca-AD CNX ("dòlars del Banc Popular Xinès")
ca-ES-valencia CNX ("dòlars del Banc Popular Xinès")
ca-FR CNX ("dòlars del Banc Popular Xinès")
ca-IT CNX ("dòlars del Banc Popular Xinès")
so ISJ ("krónaha Iceland (1918–1981)")
so-DJ ISJ ("krónaha Iceland (1918–1981)")
so-ET ISJ ("krónaha Iceland (1918–1981)")
so-KE ISJ ("krónaha Iceland (1918–1981)")
```

I would consider these bugs. The display name can be backfilled from the `one` or `other` string.

There are an additional 4660 locale/currency combinations that don't have display names, but also don't have plural rules (like [here](https://github.com/unicode-org/cldr-json/blob/0876ec40e13d54c0a6b6456392802d4de7e059cb/cldr-json/cldr-numbers-full/main/yrl-VE/currencies.json#L1502)). Those should not be generated here anyway because the map will be completely empty.

Reply:
See https://github.com/unicode-org/icu4x/pull/5399/files

Reply:
Hmm, the 4660 combinations get *generated* but they don't seem to end up in the data. They seem to be basic locales, which is why the display names are missing. I'll investigate why datagen generates them, but ignore that paragraph for now.

Reply:
Ah I guess they get fallback-deduped against `und`. In this case #5399 will speed up datagen as well.